packages/verified-fetch/src/utils/get-stream-from-async-iterable.ts

```diff
@@ -23,6 +23,11 @@ export async function getStreamFromAsyncIterable (iterator: AsyncIterable<Uint8A
     },
     async pull (controller) {
       const { value, done } = await reader.next()
+      if (options?.signal?.aborted === true) {
+        controller.error(new Error(options.signal.reason ?? 'signal aborted by user'))
+        controller.close()
+        return
+      }
```

Comment:
added in [`3b58fa0` (#26)](https://github.com/ipfs/helia-verified-fetch/pull/26/commits/3b58fa0bf5502a9e2868866a0be8accba4efc2e7).

iterator from `unixfs.cat` should handle failing this stream, but handling this explicitly may allow us to exit earlier